code/__HELPERS/mobs.dm

```diff
@@ -526,3 +526,63 @@ GLOBAL_LIST_EMPTY(species_list)
 		sleep(1)
 	if(set_original_dir)
 		AM.setDir(originaldir)
+
+///////////////////////
+///Silicon Mob Procs///
+///////////////////////
+
+//Returns a list of unslaved cyborgs
+/proc/active_free_borgs()
+	. = list()
+	for(var/mob/living/silicon/robot/R in GLOB.alive_mob_list)
+		if(R.connected_ai || R.shell)
+			continue
+		if(R.stat == DEAD)
+			continue
+		if(R.emagged || R.scrambledcodes)
+			continue
+		. += R
+
+//Returns a list of AI's
+/proc/active_ais(check_mind=FALSE, var/z = null)
+	. = list()
+	for(var/mob/living/silicon/ai/A in GLOB.alive_mob_list)
+		if(A.stat == DEAD)
+			continue
+		if(A.control_disabled)
+			continue
+		if(check_mind)
+			if(!A.mind)
+				continue
+		if(z && !(z == A.z) && (!is_station_level(z) || !is_station_level(A.z))) //if a Z level was specified, AND the AI is not on the same level, AND either is off the station...

```

Comment:
This will be a issue for multi-z stations.
I personally suggest searching for a station specific trait that will always be shared for the station.

Reply:
The reasoning I am saying its a issue is because of your check before the station specific checks.

Reply:
Tested on multi-Z donut and it works as expected. The AI is only skipped if the AI's Z level and the referenced Z level are not the same _and_ either one is not on a station Z level.